src/renderer/containers/Settings/DolphinSettings.tsx

```diff
@@ -87,12 +196,96 @@ export const DolphinSettings: React.FC<{ dolphinType: DolphinLaunchType }> = ({
           }
         />
       </SettingItem>
+
+      <SettingItem name="Gecko Codes" description="Manage and Add Gecko Codes">
+        <Button variant="outlined" color="primary" onClick={handleClickOpenGecko}>
+          Gecko Codes
+        </Button>
+      </SettingItem>
+      <Dialog open={openGecko} onClose={handleGeckoClose}>
+        <Tabs value={tabValue} onChange={handleChange}>
+          <Tab label="Add" />
+          <Tab label="Manage" />
+        </Tabs>
+        <DialogContent>
+          <select
+            id="iniPicker"
+            onChange={(e) => setIniPath(`${dolphinPath}/FM-Slippi/Sys/GameSettings/${e.target.value}`)}
+          >
+            {!iniFiles ? (
+              <option key="GALE01.ini">GALE01.ini</option>
+            ) : (
+              iniFiles.map((iniName: string, i: number) => (
+                <option key={`ini-${i}`} value={iniName}>
+                  {iniName}
+                </option>
+              ))
+            )}
+          </select>
+          <TabPanel value={tabValue} index={0}>
+            <form id="geckoForm" onSubmit={writeGeckoCodes}>
+              <TextField
+                type="text"
+                id="geckoTitle"
+                label="Gecko Name"
+                variant="outlined"
+                margin="normal"
+                onChange={({ target: { value } }) => setGeckoName(value)}
+                fullWidth
+                required
+              ></TextField>
+              <TextField
+                type="textarea"
+                id="geckoBody"
+                label="A Bunch of Numbers"
+                variant="outlined"
+                margin="normal"
+                rows="15"
+                onChange={({ target: { value } }) => setGeckoBody(value)}
+                multiline
+                fullWidth
+                required
+              ></TextField>
+              <Button type="submit" fullWidth variant="contained" color="primary">
+                Add
+              </Button>
+            </form>
+          </TabPanel>
+          <TabPanel value={tabValue} index={1}>
+            <List>
+              {!geckoCodes ? (
+                <ListItem>No Codes Found</ListItem>
+              ) : (
+                geckoCodes.map((gecko: string, i: number) => (
+                  <ListItem key={`checkbox-item-${i}`} dense>
+                    <Checkbox
+                      id={`checkbox-${i}`}
+                      checked={checked[i] === 1}
+                      disableRipple
+                      onChange={handleToggle(i)}
+                    />
+                    <ListItemText primary={gecko.substring(1)} />
+                  </ListItem>
+                ))
+              )}
+            </List>
+            <Button fullWidth onClick={loadGeckoCodes} color="primary" variant="contained">
+              Refresh
```

Comment:
I'm not a huge fan of having a refresh button. The codes should always reload when the user hits save or changes the file being looked at. The only reason it is necessary in Dolphin is because sometimes dolphin can't tell that the editor is still open and will reload before the user has saved their changes.